$OpenBSD: patch-src_od-fs_bsdsocket_posix_cpp,v 1.2 2020/07/28 20:03:33 edd Exp $

BAD_THREAD is defined as NULL, and ports-gcc doesn't want to
compare nullptr and int.

Index: src/od-fs/bsdsocket_posix.cpp
--- src/od-fs/bsdsocket_posix.cpp.orig
+++ src/od-fs/bsdsocket_posix.cpp
@@ -1403,7 +1403,7 @@ int host_sbinit (TrapContext *context, SB)
     sb->hostentsize = 1024;
 
     /* @@@ The thread should be PTHREAD_CREATE_DETACHED */
-    if (uae_start_thread ("bsdsocket", bsdlib_threadfunc, (void *)sb, &sb->thread) == BAD_THREAD) {
+    if (uae_start_thread ("bsdsocket", bsdlib_threadfunc, (void *)sb, &sb->thread) == 0) {
         write_log ("BSDSOCK: Failed to create thread.\n");
         uae_sem_destroy (&sb->sem);
         close (sb->sockabort[0]);
